### PR TITLE
Compile under Swift 4 

### PR DESCRIPTION
**Pull request by volodg**
This applies minimal changes to allow RxSwift to compile under Swift 4

### CODE REVIEW

**Comment by ldiqual** [comment]
@volodg I get this error still:

```
RxCocoa/Foundation/NSObject+Rx.swift:458:57: Value of optional type 'objc_property_t?' (aka 'Optional<OpaquePointer>') not unwrapped; did you mean to use '!' or '?'?
```

**Comment by kzaher** [comment]
Hi @volodg ,

I think there are some issues with `public typealias RxAbstractInteger = FixedWidthInteger` and `SwiftSupport/SwiftSupport.swift`.

You can try running `./scripts/all-tests.sh` locally.

File: RxCocoa/iOS/UITableView+Rx.swift [hunk 1]
```diff
@@ -168,15 +168,6 @@ extension Reactive where Base: UITableView {
 extension UITableView {
  
     /**
-    Factory method that enables subclasses to implement their own `delegate`.
-    
-    - returns: Instance of delegate proxy that wraps `delegate`.
-    */
-    public override func createRxDelegateProxy() -> RxScrollViewDelegateProxy {
```
Comment (kzaher) [inline RxCocoa/iOS/UITableView+Rx.swift]: Hi @volodg ,

I've missed these changes. I looks like these cause real issues.

Could you please explain why you've removed these `createRxDelegateProxy` method overrides?

Comment (volodg) [inline RxCocoa/iOS/UITableView+Rx.swift]: Missed it, already fixed.
"override" now forbidden inside of extension, so I moved this logic into UIScrollView

Comment (kzaher) [inline RxCocoa/iOS/UITableView+Rx.swift]: Oh crap :(

I mean, your changes are a bandaid, so they are great first step, but hoping we can find some more extensible solution :(

Comment (volodg) [inline RxCocoa/iOS/UITableView+Rx.swift]: Of cause, hope you can )

Comment (kzaher) [inline RxCocoa/iOS/UITableView+Rx.swift]: Let's try to find some better solution for this, I'll also try to figure out some better way.

**Pull request merged**